### ISSUE TITLE
Bugs... need better regex

### ISSUE THREAD

**Issue by tunnckoCore** [regexps/hex-color-regex#1]
Current

``` js
/#([a-f0-9]{6}|[a-f0-9]{3})/gi
```

and

``` js
/#([a-f0-9]{3}){1,2}/gi
```

Looks good, bug they dont catch some edge cases... like

``` js
var hex = '#7f68ZY' // should fail
var hex = '#666EFR' // should fail
```

Its easy to just set `$` or something like `(?:[^a-z0-9]+)` in the end, but this limits the global search, which for me is one of the main purposes and reasons to have that regex in the world.

@regexps/owners @johnotander thoughts?

**Comment by johno** [comment]
You should be able to use `\b` to match the end of a word:

``` javascript
 $ node
> hexReg = /#([a-f0-9]{3}){1,2}\b/gi
/#([a-f0-9]{3}){1,2}\b/gi
> hexReg.test('foo')
false
> hexReg.test('#123')
true
> hexReg.test('#123XXX')
false
> hexReg.test('#123123')
true
> hexReg.test('#1231234')
false
```

However, parsing something like minified CSS would become problematic because an ending `}` won't match:

``` javascript
> hexReg.test('#123123}')
false
```

There should probably be an `exact` option which includes the `\b` and then for the global option, one might just need to match a `#123WITHGARBAGE` as `#123`. Because, technically, it contains a valid hex color :smirk_cat:.

**Comment by tunnckoCore** [comment]
> You should be able to use \b to match the end of a word

Hm.

> one might just need to match a #123WITHGARBAGE as #123

I think against test should return false, because `valid hex color`. To be `valid` for me mean it must be 3 or 6 chars not less, not more, and starts with hash.

Yea ;d its valid `hash-tag` haha

>  because an ending } won't match

Hm.. maybe we can catch this case.

Okey, thanks, i'll try.

**Comment by tunnckoCore** [comment]
Haha this trick looks good?

```js
/#([a-f0-9]{6}|[a-f0-9]{3})(?!\})\b/gi
```

**Comment by tunnckoCore** [comment]
published as `npm@v1.0.0`

i'll open issue for the commented `}` minify problem ;d
